src/App.js

```diff
@@ -1,14 +1,42 @@
-import React from 'react'
-import axios from 'axios'
+import React, { useEffect } from 'react';
+import {
+  Routes,
+  Route,
+} from 'react-router-dom';
+import { useDispatch, useSelector } from 'react-redux';
+import { fetchData } from './components/redux/actions';
+import Home from './components/Home';
+import Paris from './components/Paris';
+import Warsaw from './components/Warsaw';
+import Barcelona from './components/Barcelona';
+import Rome from './components/Rome';
+import Lisbon from './components/Lisbon';
+import Madrid from './components/Madrid';
+import London from './components/London';
+import Brussels from './components/Brussels';
+
+import './App.css';
 
 function App() {
+  const coord = useSelector((state) => state.coord);
+  const dispatch = useDispatch();
+  useEffect(() => {
+    dispatch(fetchData());
+  }, []);
+
   return (
-    <div>
-      {
-        axios.get('http://api.openweathermap.org/data/2.5/air_pollution?lat=48.8566&lon=2.3522&appid=f1d8a4d595c7a03c5224b9f44ae09d6d').then(res=>console.log(res.data))
-      }
-    </div>
-  )
+    <Routes>
+      <Route path="/paris" element={<Paris />} />
+      <Route path="/Warsaw" element={<Warsaw />} />
+      <Route path="/Barcelona" element={<Barcelona />} />
+      <Route path="/Rome" element={<Rome />} />
+      <Route path="/Lisbon" element={<Lisbon />} />
+      <Route path="/Madrid" element={<Madrid />} />
+      <Route path="/London" element={<London />} />
+      <Route path="/Brussels" element={<Brussels />} />
+      <Route path="/" element={<Home coord={coord} />} />
```

Comment:
## [OPTIONAL]
- [ ] Nice job :clap: on navigating between pages without reloading. However, I have seen all of your components(Barcelona, London, Madrid,...) have the same structure. The better approach I can suggest is to use reusable components as this is one of the beauties of `Reactjs`.

Reply:
cool